server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -236,6 +236,9 @@ static boolean execute(SearchContext searchContext,
                     System.arraycopy(oldFormats, 0, newFormats, 1, oldFormats.length);
                     sortAndFormatsForRewrittenNumericSort = searchContext.sort(); // stash SortAndFormats to restore it later
                     searchContext.sort(new SortAndFormats(new Sort(newSortFields), newFormats));
+                    if (LOGGER.isTraceEnabled()) {
+                        LOGGER.trace("Sort optimization on the field [" + oldSortFields[0].getField() + "] was enabled!");
+                    }
```

Comment:
Not sure that this helps the debugging ;)

Reply:
@jimczi What would be the way to see if the optimization was used? LOGGER.trace is a not a good way?